serf/serf.go

```diff
@@ -1127,26 +1131,44 @@ func (s *Serf) handleNodeLeaveIntent(leaveMsg *messageLeave) bool {
 		}
 
 		if leaveMsg.Prune {
-			s.logger.Printf("[INFO] serf: EventMemberReap (forced): %s %s", member.Name, member.Member.Addr)
-			s.leftMembers = removeOldMember(s.leftMembers, member.Name)
-			s.eraseNode(member)
+			s.handlePrune(member)
 		}
 
 		return true
 
-	case StatusLeft:
+	case StatusLeaving:
```

Comment:
You don't have to do this, but I'm just noting that since these two cases have the same bodies you could change it to:
```
case StatusLeaving, StatusLeft:
   ...common body...
```